### PR TITLE
Update Event Bus architecture

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Update the Eventing architecture diagrams and descriptions
**Related issue(s)**
See #3563

### CODE REVIEW

File: docs/event-bus/docs/02-01-event-bus.md [hunk 1]
```diff
@@ -4,25 +4,32 @@ title: Architecture
...
 ## Event flow
 
-1. The external solution integrated with Kyma makes a REST API call to the Application Connector to indicate that a new Event is available.
-
-2. The Application Connector enriches the Event with the details of its source.
+### Configuration
 
-> **NOTE:** There is always one dedicated instance of the Application Connector for every instance of an external solution connected to Kyma.
+1. A user configures a lambda or a serverless application to be triggered by an Event from an external solution.
+> **NOTE**: While the system automatically creates a KymaSubscription resource for a lambda, the user must create one manually for a microservice.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
> **NOTE**: While the system automatically creates a Kyma Subscription resource for a lambda, the user must create one manually for a microservice.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 2]
```diff
@@ -4,25 +4,32 @@ title: Architecture
...
+2. The `subscription-controller-knative` reacts on the creation of the KymaSubscription. After verifying that the Event was activated, it creates a KnativeChannel and a KnativeSubscription.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
2. The `subscription-controller-knative` reacts on the creation of the Kyma Subscription. After verifying that the Event was activated, it creates a Knative Channel and a Knative Subscription.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 3]
```diff
@@ -4,25 +4,32 @@ title: Architecture
...
+3. The NATS controller reacts on the creation of a Knative Channel and creates the required Kubernetes and Istio services.
+4. The NATS dispatcher reacts on the creation of a KnativeSubscription and creates the NATS Streaming subscription.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
4. The NATS dispatcher reacts on the creation of a Knative Subscription and creates the NATS Streaming subscription. 
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 4]
```diff
@@ -4,25 +4,32 @@ title: Architecture
...
+3. The NATS controller reacts on the creation of a Knative Channel and creates the required Kubernetes and Istio services.
+4. The NATS dispatcher reacts on the creation of a KnativeSubscription and creates the NATS Streaming subscription. 
 
-3. The Application Connector makes a REST API call to `publish` and sends the enriched Event.
 
-4. `publish` saves the information in the NATS Streaming database.
+### Runtime
 
-5. NATS Streaming stores the Event details in the Persistence storage volume to ensure the data is not lost if the NATS Streaming crashes.
+1. The external solution integrated with Kyma makes a REST API call to the Application Connector to indicate that a new Event is available.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]: we should mark this as a prerequisite.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 5]
```diff
@@ -4,25 +4,32 @@ title: Architecture
...
+2. The Application Connector enriches the Event with the details of its source.
 
-6. If the Subscription [validation process](#architecture-architecture-event-validation) completes successfully, `push` consumes the Event from NATS Streaming.
+> **NOTE:** There is always one dedicated instance of the Application Connector for every instance of an external solution connected to Kyma.
 
-7. `push` delivers the Event to the lambda or the service.
+3. The Application Connector makes a REST API call to PublishKnative and sends the enriched Event.
+4. PublishKnative forwards the Event to the relevant knative-channel service URL which is inferred based on **source id** , **event type** and **event type version** parameters.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
4. `publish-knative` forwards the Event to the relevant Knative Channel service URL which is inferred based on **source id** , **event type** and **event type version** parameters.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 6]
```diff
@@ -40,10 +47,10 @@ Before the Event Bus forwards the Event to the receiver, the sub-validator perfo
...
 
-![Event validation process](./assets/event-validation.png)
+![Event validation process](./assets/event-validation.svg)
 
 1. The Kyma user defines a lambda or a service.
 2. The Kyma user creates a Subscription custom resource.
-3. The sub-validator reads the new Subscription.
-4. The sub-validator refers to the EventActivation resource to check if the Event in the Subscription is activated for the given Namespace.
-5. The sub-validator updates the Subscription resource accordingly.
+3. The SubscriptionController reads the new Subscription.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
3. The `subscription-controller-knative` reads the new Subscription.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 7]
```diff
@@ -40,10 +47,10 @@ Before the Event Bus forwards the Event to the receiver, the sub-validator perfo
...
+4. The SubscriptionController refers to the EventActivation resource to check if the Event in the Subscription is activated for the given Namespace.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
4. The `subscription-controller-knative` refers to the Event Activation resource to check if the Event Type in the Subscription is activated for the given Namespace.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 8]
```diff
@@ -40,10 +47,10 @@ Before the Event Bus forwards the Event to the receiver, the sub-validator perfo
...
+5. The SubscriptionController updates the Subscription resource accordingly.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
5. The `subscription-controller-knative` updates the Subscription resource accordingly.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 9]
```diff
@@ -2,33 +2,42 @@
 title: Architecture
 ---
 
-See the diagram and steps for an overview of the basic Event Bus flow:
+See the diagram overview of the Event flow configuration and Event cons
 
-![Event Bus architecture](./assets/event-bus-architecture.png)
+![Configure and Consume Events](./assets/configure-consume-events.svg)
```
[the remark this answers is not in the document]
Comment (abbi-gaurav) [reply docs/event-bus/docs/02-01-event-bus.md]: I would suggest starting with the same color. It seems simpler to me in that case.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 10]
```diff
@@ -2,33 +2,42 @@
...
 
-## Event flow
 
-1. The external solution integrated with Kyma makes a REST API call to the Application Connector to indicate that a new Event is available.
+## Eventing configuration 
 
-2. The Application Connector enriches the Event with the details of its source.
+1. A user configures a lambda or a serverless application to be triggered by an Event from an external solution. 
+>**NOTE**: For a serverless application the user must create a Kyma Subscription resource manually. For a lambda, the it is created automatically.
```
[the remark this answers is not in the document]
Comment (abbi-gaurav) [reply docs/event-bus/docs/02-01-event-bus.md]: The diagram should somehow depict that it belongs to event consumption and configuration. Currently, it precedes that section

**Comment by bszwarc** [comment]
@abbi-gaurav , I added some more improvements - you can have a look :)

File: docs/event-bus/docs/02-01-event-bus.md [hunk 11]
```diff
@@ -2,48 +2,53 @@
 title: Architecture
 ---
 
-See the diagram and steps for an overview of the basic Event Bus flow:
+When you create a lambda or a service to perform a given business functionality, you must define which Events trigger it. Define triggers by creating the Subscription CR in which you instruct the Event Bus to forward the Events of a particular type to your lambda. 
+For example, whenever the `order-created` Event comes in, the Event Bus consumes it by saving it in NATS Streaming and persistence. Then it sends it to the receiver specified in the Subscription definition.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
For example, whenever the `order-created` Event comes in, the Event Bus stores it in NATS Streaming. The event is dispatched to the receiver specified in the Subscription definition.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 12]
```diff
@@ -2,48 +2,53 @@
 title: Architecture
 ---
 
-See the diagram and steps for an overview of the basic Event Bus flow:
+When you create a lambda or a service to perform a given business functionality, you must define which Events trigger it. Define triggers by creating the Subscription CR in which you instruct the Event Bus to forward the Events of a particular type to your lambda.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]: We could possibly move this inside Event consumption as the context is closer to that.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 13]
```diff
@@ -2,48 +2,53 @@
...
 
-> **NOTE:** There is always one dedicated instance of the Application Connector for every instance of an external solution connected to Kyma.
+![Configure and Consume Events](./assets/configure-consume-events.svg)
 
-3. The Application Connector makes a REST API call to `publish` and sends the enriched Event.
 
-4. `publish` saves the information in the NATS Streaming database.
+1. A user creates a lambda or a service to be triggered by an Event coming from an external solution. 
+    >**NOTE**: For a service the user must create a Kyma Subscription resource manually. For a lambda, it is created automatically.
+2. The **subscription-controller-knative** component reacts on the creation of Kyma Subscription. It [verifies](#event-validation) if the Event type has subscription permissions to if so, it creates the Knative Channel and Knative Subscription resources.
```
Comment (abbi-gaurav) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
2. The **subscription-controller-knative** component reacts on the creation of Kyma Subscription. It [verifies](#event-validation) if the Event type from the application can be consumed in the namespace in which Kyma Subscription has been created, it creates the Knative Channel and Knative Subscription resources.
```

File: docs/event-bus/docs/02-01-event-bus.md (hunk 12 above)
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Moved:)

File: docs/event-bus/docs/02-01-event-bus.md (hunk 13 above)
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 14]
```diff
@@ -2,48 +2,51 @@
...
 
-> **NOTE:** The Event Bus creates a separate Event Trigger for each Subscription.
+## Event validation 
 
-Before the Event Bus forwards the Event to the receiver, the sub-validator performs a security check to verify the permissions for this Event in a given Namespace. It reads all new Subscription resources and refers to the EventActivation resource to check whether a particular Event type is enabled in a given Namespace. If the Event is enabled for an Namespace, it updates the Subscription resource with the information. Based on the information, `push` sends the Event to the lambda or the service.
+The **subscription-controller-knative** checks if the Namespace can receive Events from the Application. It performs the check for each Kyma Subscription created in a Namespace for a particular Event type with a version for a specific Application.
```
Comment (kazydek) [inline docs/event-bus/docs/02-01-event-bus.md]:
```suggestion
**subscription-controller-knative** checks if the Namespace can receive Events from the Application. It performs the check for each Kyma Subscription created in a Namespace for a particular Event type with a version for a specific Application.
```

File: docs/event-bus/docs/02-01-event-bus.md [hunk 15]
```diff
@@ -2,48 +2,51 @@
 title: Architecture
 ---
 
-See the diagram and steps for an overview of the basic Event Bus flow:
+## Event consumption
 
-![Event Bus architecture](./assets/event-bus-architecture.png)
+When you create a lambda or a service to perform a given business functionality, you must define which Events trigger it. Define triggers by creating the [Subscription CR](components/event-bus/#custom-resource-subscription) where you instruct the Event Bus to forward the Events of a particular type to your lambda. 
+For example, whenever the `order-created` Event comes in the Event Bus stores it in NATS Streaming. It then dispatches it to the receiver specified in the Subscription CR definition.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 16]
```diff
@@ -2,48 +2,51 @@
...
 
-## Event flow
-
-1. The external solution integrated with Kyma makes a REST API call to the Application Connector to indicate that a new Event is available.
-
-2. The Application Connector enriches the Event with the details of its source.
+> **NOTE:** The Event Bus creates a separate Event Trigger for each Subscription.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 17]
```diff
@@ -2,48 +2,51 @@
...
 
-> **NOTE:** There is always one dedicated instance of the Application Connector for every instance of an external solution connected to Kyma.
+![Configure and Consume Events](./assets/configure-consume-events.svg)
 
-3. The Application Connector makes a REST API call to `publish` and sends the enriched Event.
 
-4. `publish` saves the information in the NATS Streaming database.
+1. A user creates a lambda or a service to be triggered by an Event coming from an external solution.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 18]
```diff
@@ -2,48 +2,51 @@
...
 
-> **NOTE:** There is always one dedicated instance of the Application Connector for every instance of an external solution connected to Kyma.
+![Configure and Consume Events](./assets/configure-consume-events.svg)
 
-3. The Application Connector makes a REST API call to `publish` and sends the enriched Event.
 
-4. `publish` saves the information in the NATS Streaming database.
+1. A user creates a lambda or a service to be triggered by an Event coming from an external solution. 
+    >**NOTE**: For a service, the user must create a Kyma Subscription resource manually. For a lambda, it is created automatically.
+2. The **subscription-controller-knative** component reacts on the creation of Kyma Subscription.  It [verifies](#event-validation) if the Event type from the Application can be consumed in the Namespace where the Kyma Subscription has been created.  If so, it creates the Knative Channel and Knative Subscription resources.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: application :)

File: docs/event-bus/docs/02-01-event-bus.md [hunk 19]
```diff
@@ -2,48 +2,51 @@
...
-7. `push` delivers the Event to the lambda or the service.
+1. The external application integrated with Kyma makes a REST API request to the Application Connector's Events Gateway to indicate that a new Event is available. The request provides the Application Connector with the Event metadata. 
+2. The Application Connector enriches the Event with the details of its source.
 
-## Event validation
+    > **NOTE:** There is always one dedicated instance of the Application Connector for every instance of an external solution connected to Kyma.
 
-The Event Bus performs Event validation before it allows Event consumption.
+3. The Application Connector makes a REST API call to the **publish-knative** component and sends the enriched Event.
+4. **publish-knative** makes the HTTP payload compatible with Knative and sends the Event to the relevant **knative-channel** service URL which is inferred based on **source id**, **event type** and **event type version** parameters.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 20]
```diff
@@ -2,48 +2,51 @@
...
+5. Istio Virtual Service forwards the Event further to the **nats-dispatcher** service served by the **nats-dispatcher** Pod.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 21]
```diff
@@ -2,48 +2,51 @@
...
+6. The NATS dispatcher stores the Event in NATS Streaming which stores the Event details in the Persistence storage volume.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: correct, should be nats-dispatcher.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 22]
```diff
@@ -2,48 +2,51 @@
...
 
-3. The Application Connector makes a REST API call to `publish` and sends the enriched Event.
 
-4. `publish` saves the information in the NATS Streaming database.
+1. A user creates a lambda or a service to be triggered by an Event coming from an external solution. 
+    >**NOTE**: For a service, the user must create a Kyma Subscription resource manually. For a lambda, it is created automatically.
+2. The **subscription-controller-knative** component reacts on the creation of Kyma Subscription.  It [verifies](#event-validation) if the Event type from the Application can be consumed in the Namespace where the Kyma Subscription has been created.  If so, it creates the Knative Channel and Knative Subscription resources.
+3. The **nats-controller** reacts on the creation of a Knative Channel and creates the required Kubernetes and Istio services.
+4. The **nats-dispatcher** reacts on the creation of a Knative Subscription and creates the NATS Streaming subscription. 
+5. The **nats-dispatcher** picks the Event and dispatches it to the configured lambda or the service URL as an HTTP POST request. The lambda reacts on the received Event.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md (hunk 14 above)
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 23]
```diff
@@ -2,48 +2,51 @@
...
-![Event validation process](./assets/event-validation.png)
+![Event validation process](./assets/event-validation.svg)
 
 1. The Kyma user defines a lambda or a service.
 2. The Kyma user creates a Subscription custom resource.
-3. The sub-validator reads the new Subscription.
-4. The sub-validator refers to the EventActivation resource to check if the Event in the Subscription is activated for the given Namespace.
-5. The sub-validator updates the Subscription resource accordingly.
+3. The **subscription-controller-knative** reads the new Subscription.
+4. The **subscription-controller-knative** reads the EventActivation CR to verify if it exists in the Namespace for a certain Application.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 24]
```diff
@@ -2,48 +2,51 @@
...
+5. The **subscription-controller-knative**  updates the Subscription resource accordingly with the activation status `true` or `false`. The Event Bus uses this status to allow or prohibit Event delivery.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 25]
```diff
@@ -2,48 +2,51 @@
...
 
-![Event validation process](./assets/event-validation.png)
+![Event validation process](./assets/event-validation.svg)
 
 1. The Kyma user defines a lambda or a service.
 2. The Kyma user creates a Subscription custom resource.
-3. The sub-validator reads the new Subscription.
-4. The sub-validator refers to the EventActivation resource to check if the Event in the Subscription is activated for the given Namespace.
-5. The sub-validator updates the Subscription resource accordingly.
+3. The **subscription-controller-knative** reads the new Subscription.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 26]
```diff
@@ -2,48 +2,51 @@
...
-> **NOTE:** There is always one dedicated instance of the Application Connector for every instance of an external solution connected to Kyma.
+![Configure and Consume Events](./assets/configure-consume-events.svg)
 
-3. The Application Connector makes a REST API call to `publish` and sends the enriched Event.
 
-4. `publish` saves the information in the NATS Streaming database.
+1. A user creates a lambda or a service to be triggered by an Event coming from an external solution. 
+    >**NOTE**: For a service, the user must create a Kyma Subscription resource manually. For a lambda, it is created automatically.
+2. The **subscription-controller-knative** component reacts on the creation of Kyma Subscription.  It [verifies](#event-validation) if the Event type from the Application can be consumed in the Namespace where the Kyma Subscription has been created.  If so, it creates the Knative Channel and Knative Subscription resources.
+3. The **nats-controller** reacts on the creation of a Knative Channel and creates the required Kubernetes and Istio services.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: Done.

File: docs/event-bus/docs/02-01-event-bus.md [hunk 27]
```diff
@@ -2,48 +2,51 @@
...
+4. The **nats-dispatcher** reacts on the creation of a Knative Subscription and creates the NATS Streaming subscription.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/event-bus/docs/02-01-event-bus.md]: It will be Subscription.

**Pull request merged**